common/tools/cointool.py

```diff
@@ -742,9 +767,16 @@ def dump(
     You can also specify filters, in the form '-f field=value' (or '-F' for inverse
     filter). Filter values are case-insensitive and support shell-style wildcards,
     so '-f name=bit*' finds all coins whose names start with "bit" or "Bit".
+
+    Also devices can be used as filters. For example to find out which coins are
+    supported in Suite and connect but not on Trezor 1, it is possible to say
+    '-d suite -d connect -D trezor1'.
+
+    Includes even the wallet data, unless turned off by '-W'.
+    These can be filtered by using '-f', for example `-f 'wallet=*exodus*'` (* are necessary)
     """
     if exclude_tokens:
-        exclude_type = ("erc20",)
+        exclude_type += ("erc20",)
```

Comment:
Also, the reason to put `list[str]` is that when we have a tuple, it is a sign that we are not changing these. And in the code, we are two times changing some of those values - `exclude_type += ("erc20",)` and `exclude += ("address_bytes",)`.

However, type-checker is fine with these, even when all is marked as `tuple[str, ...]`

Reply:
hah, interestingly, this works fine:
```python
a = []
a += (1,)
```
so that's why the typechecker doesn't complain

> the reason to put list[str] is that when we have a tuple, it is a sign that we are not changing these. And in the code, we are two times changing some of those values

yeah but given that we are actually receiving tuples, we're technically not allowed to modify them.
(i'd call this a bad decision on the side of Click but well 🤷‍♀️ )

The right type annotation would be `Sequence[str]`, which indicates that it's list-like but we're not really allowed to do anything to it. But then the correct way to change the values would be `val = list(val) + ["foo"]` which ... feels like we're jumping through hoops for the typechecker's sake.
So practically in terms of "code is documentation", it seems best to really do `tuple[str, ...]` for the annotation.

Reply:
> So practically in terms of "code is documentation", it seems best to really do `tuple[str, ...]` for the annotation.

In an individual case probably yes, but from consistency perspective, we would "need/want" to change it to `tuple[str, ...]` in all the places in our code where we already use `list[str]` as input from `click`.

Therefore my vote is for leaving `list[str]` there.

(also, is it really important to be appending tuples here, cannot we just do 
```python
exclude_type += ["erc20"]
exclude += ["address_bytes"]
```
or 
```python
exclude_type.append("erc20")
exclude.append("address_bytes")
```
?)

Reply:
Sorry, the last question does not make sense, both of those are generating a Runtime error

Reply:
exactly, but the `list[]` annotation suggests that this should be possible.

i'd just fix the annotation here -- i don't think there are many places where we're accepting a list...?

Reply:
> i'd just fix the annotation here -- i don't think there are many places where we're accepting a list...?

Indeed, I was thinking we are accepting the list more often, turns out not very much. Changed to `tuple[str, ...]` in [3f47380](https://github.com/trezor/trezor-firmware/pull/2286/commits/3f47380c5d9dde06c2d8ac13261a1a74b19ed7c8)